Config-Files/C&C++/Windows/tasks.json

```diff
@@ -1,54 +1,59 @@
-//Author : 5hfT
-//date:11th Feb,2020
+//Author : 5hft
+//Modifier : Mahdi
+//date:28th June,2022
 //Task:shortcut to run a cpp file which will take input and show output of the programme automatically with programme runtime status such as memory comsume and execution time
 {
-  "version": "2.0.0",
-  "tasks": [
-    {
-      "label": "Compile and run",
-      "type": "shell",
-      "command": "",
-      "args": [
-        "copy",
-        "${file}",
-        "${workspaceFolder}\\jspwTest.cpp",
-        "&&",
-        "g++",
-        "jspwTest.cpp",
-        "-o",
-        "jspwTest",
-        "&&",
-        "jspwTest",
-        "<",
-        "input.txt",
-        ">",
-        "output.txt",
-        "&&",
-        "del",
-        "jspwTest.exe",
-        "&&",
-        "del",
-        "jspwTest.cpp"
-      ],
-      "group": {
-        "kind": "build",
-        "isDefault": true,
-      },
-      "problemMatcher": {
-        "owner": "cpp",
-        "fileLocation": [
-          "relative",
-          "${workspaceRoot}"
+    "version": "2.0.0",
+    "tasks": [
+      {
+        "label": "Compile and run",
+        "type": "shell",
+        "command": "",
+        "args": [
+          "cp",
+          "'${file}'",
+          "'${workspaceFolder}\\${fileBasename}'",
+          "&&",
+          "fileName=\"${fileBasename}\"",
+          "&&",
+          "objectFile=\"${fileName%.*}.out\"",          
+          "&&",
+          "g++",
+          "'${fileBasename}'",
+          "-o",
+          "${objectFile}",
+          "&&",
+          "./${objectFile}",
+          "<",
+          "input.txt",
+          ">",
+          "output.txt",
+          "&&",
+          "rm",
+          "${objectFile}",
+          "&&",
+          "rm",
+          "${fileBasename}",

```

Comment:
did you just removed basefile ? @Mahdi-Hasan

Reply:
The base file was copied to working directory. I just removing the copied file from working directory. But the main file is remined in its own folder. @jspw